server/models/groups.js

```diff
@@ -1,50 +1,94 @@
+/* eslint-disable no-restricted-syntax */
 const { HttpError } = require('../utils/errors');
 const prisma = require('../db');
 
 class Group {
-  constructor(name, id) {
-    this.name = name;
-    this.id = id;
-  }
-
-  async isDuplicate() {
-    const { id, name } = this;
-    const record = await prisma.Group.findUnique({
-      where: {
-        name
-      }
-    });
-
-    if (record) {
-      if (record.groupAdmin === id) {
-        throw new HttpError(409, 'Group by user already exists');
-      }
+    constructor(name, id) {
+        this.name = name;
+        this.id = id;
+    }
+
+    async isDuplicate() {
+        const { id, name } = this;
+        const record = await prisma.Group.findUnique({
+            where: {
+                name,
+            },
+        });
+
+        if (record) {
+            if (record.groupAdmin === id) {
+                throw new HttpError(409, 'Group by user already exists');
+            }
+        }
+    }
+
+    async createGroup() {
+        const { id, name } = this;
+
+        await prisma.Group.create({
+            data: {
+                name,
+                groupAdmin: id,
+            },
+        });
+
+        const group = await prisma.$queryRaw`SELECT "id" FROM "Group" WHERE "name" = ${name} AND "groupAdmin" = ${id}`;
+
+        await prisma.GroupMembership.create({
+            data: {
+                group_id: group[0].id,
+                user_id: id,
+            },
+        });
+    }
+
+    static async isGroupAdmin(groupId, userId) {
+        const record = await prisma.Group.findUnique({
+            where: {
+                id: Number(groupId),
+            },
+        });
+        if (record.groupAdmin !== userId) {
+            throw new HttpError(403, 'Only the Group administrator can add users');
+        }
+    }
+    
+    static async sanitizeData(arr, groupId) {
```

Comment:
since we are taking in raw data, the user might input user_ids which already exists, the idea is to sieve out those who already exists and add only those users who do not